src/components/pickers/Menu/MenuItem.tsx

```diff
@@ -82,6 +83,7 @@ export function MenuItem<T>(props: MenuItemProps<T>) {
       <StyledMenuItem
         {...mergeProps(menuItemProps, hoverProps)}
         ref={ref}
+        data-qa={itemProps.qa ? `MenuItem-${itemProps.qa}` : `MenuItem-${key}`}
```

Comment:
I'm not sure it's a good idea to override a user-specified `qa` attribute

Reply:
`itemProps` is a props of `MenuButton`, also this code share qa with `li` item
When you set `qa` prop in `<Menu.Item key="foo" qa="foo-qa">....` you got 
`<li data-qa"MenuItem-foo-qa"><button data-qa="foo-qa">`